core/station/impl/src/services/external_canister.rs

```
@@ -517,196 +520,246 @@ impl ExternalCanisterService {
-                        ));
+        // manage the policies for the `Call` requests
+        if let Some(updated_call_policies) = input.calls {
+            match &updated_call_policies {
+                ExternalCanisterChangeCallRequestPoliciesInput::ReplaceAllBy(calls) => self
+                    .maybe_mutate_canister_calls_request_policies(
+                        external_canister,
+                        &current_policies.calls,
+                        calls,
+                    )?,
+                ExternalCanisterChangeCallRequestPoliciesInput::OverrideSpecifiedByExecutionMethods(
+                    calls,
+                ) => {
+                    // first aggregates the calls by the execution method
+                    let calls_by_execution_method = calls.iter().fold(
+                        HashMap::new(),
+                        |mut acc: HashMap<
+                            String,
+                            Vec<ExternalCanisterCallRequestPolicyRuleInput>,
+                        >,
+                         call| {
+                            acc.entry(call.execution_method.clone())
+                                .or_default()
+                                .push(call.clone());
+                            acc
+                        },
+                    );
 
-                        continue;
+                    for (method, calls) in calls_by_execution_method {
+                        self.maybe_mutate_canister_calls_request_policies(
+                            external_canister,
+                            &self
+                                .request_policy_repository
+                                .find_external_canister_call_policies_by_execution_method(
+                                    &external_canister.canister_id,
+                                    &method,
+                                ),
+                            &calls,
+                        )?;
                     }
```

same here: we should add this back and ideally fail the request instead of continuing